agent/blacklist_test.go

```diff
@@ -0,0 +1,52 @@
+package agent
+
+import (
+	"testing"
+)
+
+func TestBlacklist(t *testing.T) {
+	type pathCase struct {
+		path string
+		want bool
+	}
+
+	tests := []struct {
+		desc     string
+		prefixes []string
+		paths    []pathCase
+	}{
+		{
+			"nothing disallowed",
+			nil,
+			[]pathCase{
+				{"/", false},
+				{"/v1/agent/self", false},
+			},
+		},
+		{
+			"prefix match",
+			[]string{
+				"/v1/acl",
+				"/v1/agent/self",
+			},
+			[]pathCase{
+				{"/", false},
```

Comment:
I'd use generic endpoint names like `/a`, `/b` to make it clear what you're testing. You're covering `/a`, `/aa` and `/a/b` for the `/a` blacklist but that isn't obvious. Also, with the two entries you didn't cover `/v1/acl`